peps/pep-0776.rst

```diff
@@ -0,0 +1,794 @@
+PEP: 776
+Title: Emscripten Support
+Author: Hood Chatham <roberthoodchatham at gmail.com>
+Sponsor: Łukasz Langa <lukasz at python.org>
+Discussions-To: https://discuss.python.org/t/84996
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 18-Mar-2025
+Python-Version: 3.14
+Post-History: `18-Mar-2025 <https://discuss.python.org/t/84996>`__
+
+Abstract
+========
+
+`Emscripten <https://emscripten.org/>`__ is a complete open source compiler
+toolchain which compiles C/C++ code into WebAssembly/JavaScript executables that
+run in JavaScript runtimes including browsers and Node.js.
+
+This PEP formalizes the addition of Tier 3 for Emscripten support in Python 3.14
+which `was approved by the steering council on October 25, 2024
+<https://github.com/python/steering-council/issues/256>`__. The goal is to allow
+Pyodide wheels to be uploaded to PyPI.
+
+Motivation
+==========
+
+A web browser is a universal computing platform, available on Windows, macOS,
+Linux, and every smartphone.
+
+`The Pyodide project <https://pyodide.org/>`__ has for years supported
+Emscripten Python. Hundreds of thousands of students have learned Python through
+Pyodide via projects like `Capytale
+<https://web.archive.org/web/20241211090946/https://cfp.jupytercon.com/2023/talk/TJ9YEV/>`__
+and `PyodideU <https://stanford.edu/~cpiech/bio/papers/pyodideU.pdf>`__. Pyodide
+is also increasingly being used by Python packages to provide interactive
+documentation. This demonstrates both the importance and the maturity of the
+Emscripten platform.
+
+Emscripten and WASI are also the only supported platforms that offer any
+meaningful sandboxing.
+
+Goals
+=====
+
+It is our long term goal to upstream the entire Pyodide runtime into CPython,
+but this is out of scope for the present PEP. This PEP only attempts to
+establish the foundation for future work.
+
+Runtime Goals
+-------------
+
+1. To describe the current state of the CPython Emscripten runtime
+2. To describe the current state of the Pyodide runtime
+3. To identify minor features to be upstreamed from the Pyodide runtime into the
+   CPython Emscripten runtime
+
+The minor features identified here are all features that could be implemented
+without a PEP. We discuss more significant runtime features that we would like
+to implement but we defer decisions on those features to subsequent PEPs.
+
+Packaging Goals
+---------------
+
+1. To describe Pyodide's packaging tooling
+2. To describe Pyodide's wheel abi to a similar level of detail as the manylinux
+   PEPs
+3. To request that Pyodide wheels be allowed for upload to PyPI
```

Comment:
A PEP specifies, rather than requesting -- if this PEP is accepted and implemented, Pyodide wheels will be permitted:

```suggestion
3. For Pyodide wheels to be allowed for upload to PyPI
```